components/cice/cime_config/buildnml

```diff
@@ -70,9 +70,10 @@ def buildnml(case, caseroot, compname):
                                  from_dir=os.path.join(srcroot, "components/cice/bld")).split()
 
         if int(config[0]) >= 0:
-            case.set_value("CICE_BLCKX",         config[2])
-            case.set_value("CICE_BLCKY",         config[3])
-            case.set_value("CICE_MXBLCKS",       config[4])
+            cice_blckx, cice_blcky, cice_mxblcks = config[2:5]
```

Comment:
The bug is here. The original code has the potential for case's CICE_BLCKX, CICE_BLCKY, and CICE_MXBLCKS to become out-of-sync with the local variables cice_blckx, etc which are used to define CICE_CPPDEFS below.